src/app/pages/system/view-enclosure/stores/enclosure-store.service.ts

```diff
@@ -0,0 +1,481 @@
+} from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { PoolTopologyCategory } from 'app/enums/pool-topology-category.enum';
+import { TopologyItemType } from 'app/enums/v-dev-type.enum';
+import { ApiEvent } from 'app/interfaces/api-message.interface';
+import {
+  Enclosure,
+  EnclosureElement,
+  EnclosureElementsGroup,
+  EnclosureSlot,
+  EnclosureView,
+} from 'app/interfaces/enclosure.interface';
+import { Pool } from 'app/interfaces/pool.interface';
+import {
+  Disk, TopologyDisk, TopologyItem, VDev,
+} from 'app/interfaces/storage.interface';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { DialogService, StorageService, WebSocketService } from 'app/services';
+
+export interface EnclosureState {
+  areEnclosuresLoading: boolean;
+  arePoolsLoading: boolean;
+  areDisksLoading: boolean;
+  enclosures: Enclosure[];
+  pools: Pool[];
+  disks: Disk[];
+  enclosureViews?: EnclosureView[];
+  selectedEnclosure?: number | null;
+}
+
+const initialState: EnclosureState = {
+  areEnclosuresLoading: false,
+  arePoolsLoading: false,
+  areDisksLoading: false,
+  enclosures: [],
+  pools: [],
+  disks: [],
+  enclosureViews: [],
+  selectedEnclosure: 0,
+};
+
+interface ProcessParameters {
+  pools: Pool[];
+  disks: Disk[];
+  enclosures: Enclosure[];
+}
+
+@Injectable()
+export class EnclosureStore extends ComponentStore<EnclosureState> {
+  readonly enclosures$ = this.select((state) => state.enclosures);
+  readonly pools$ = this.select((state) => state.pools);
+  readonly areEnclosuresLoading$ = this.select((state) => state.areEnclosuresLoading);
+  readonly arePoolsLoading$ = this.select((state) => state.arePoolsLoading);
+  readonly areDisksLoading$ = this.select((state) => state.areDisksLoading);
+  readonly disks$ = this.select((state) => state.disks);
+  readonly data$ = this.select((state) => state);
+  readonly enclosureViews$ = this.select((state) => state.enclosureViews);
+  readonly selectedEnclosure$ = this.select((state) => state.selectedEnclosure);
+
+  private defaultEnclosureSelection = 0;
+  private selectedSlotNumber: number | null = null;
+
+  constructor(
+    private ws: WebSocketService,
+    private dialogService: DialogService,
+    private sorter: StorageService,
+  ) {
+    super(initialState);
+  }
+
+  readonly loadData = this.effect((triggers$: Observable<void>) => {
+    return triggers$.pipe(
+      tap(() => {
+        this.patchState({
+          ...initialState,
+          areEnclosuresLoading: true,
+          arePoolsLoading: true,
+          areDisksLoading: true,
+        });
+      }),
+      switchMap(() => this.updateState()),
+    );
+  });
+
+  updateState(): Observable<{
+    enclosures: Enclosure[];
+    pools: Pool[];
+    disks: Disk[];
+    enclosureViews: EnclosureView[];
+  }> {
+    return forkJoin({
+      enclosures: this.getEnclosures().pipe(
```

Comment:
I would just simplify to private `getEnclosures` that would have necessary logic to update state/log errors in it.